### DISCUSSION TITLE
Read nodal thickness values in ABAQUS input files

### DISCUSSION THREAD

**Discussion by yw5aj** [libMesh/libmesh#3567] [Q&A] [answered]
Hi all,

I'm trying to work with `abaqus_io.C` to read nodal thickness from the mesh file and store appropriately for later assembly of the shell element stiffness matrix. The format roughly look like this:

```
*Nodal Thickness
1,   1.9876543210e+00
2,   1.7123488963e+00
3,   1.1234520861e+00
4,   1.1239897738e+00
5,   1.1234567890e+00
```

Reading the values seem straight-forward, but I wonder if I can get help on where I should appropriately store those values?

Thanks,
Shawn

**Comment by roystgnr** [comment, marked as answer]
`your_mesh.add_node_datum<Real>("nodal_thickness")`, perhaps?  That will allocate a floating-point value for every node.  The main disadvantage is that you can't naturally store superparameteric data that way, but if you just have one value per node to begin with then that's probably fine in your case.

**Reply by yw5aj** [reply]
Thanks a lot Roy! That makes a lot of sense.
